GeneratorInterface/Pythia8Interface/src/ResonanceDecayFilterHook.cc

```diff
@@ -28,7 +30,13 @@ bool ResonanceDecayFilterHook::initAfterBeams() {
     if ( (did == 14 || did == 16) && allNuAsEquivalent_) {
       did = 12;
     }
-    
+    if ( (did == 2 || did==3 || did==4 || did==5 ) && udscAsEquivalent_) {
+      did = 1;
+    }
+    if ( (did == 2 || did==3 || did==4 || did==5 || did==6) && udscbAsEquivalent_) {
+      did = 1;
+    }
+
```

Comment:
udsc are 1,2,3,4 (not in that order exactly)
and udscb are 1,2,3,4,5

6 is the top quark :)